comp/core/gui/guiimpl/agent.go

```diff
@@ -91,17 +91,19 @@ func getVersion(w http.ResponseWriter, _ *http.Request) {
 }
 
 // Sends the agent's hostname
-func getHostname(w http.ResponseWriter, r *http.Request) {
-	hname, e := hostname.Get(r.Context())
-	if e != nil {
-		log.Errorf("Error getting hostname: %s", e.Error())
-		w.Write([]byte("Error: " + e.Error()))
-		return
-	}
+func getHostname(hostname hostnameinterface.Component) func(http.ResponseWriter, *http.Request) {
```

Comment:
Both `comp/api/commonendpoints/impl/common_endpoints.go` and this file has this exact same function. It would be nice to combine them into a common location, though I don't have an immediate suggestion for where that would be.

Reply:
I thought the same, I'll look into it once this is merged 👍